Removing Local Extension crash directus with EXTENSIONS_AUTO_RELOAD

### Describe the Bug

When removing [Local Extension](https://docs.directus.io/extensions/creating-extensions.html#local-extension) from the disk with `EXTENSIONS_AUTO_RELOAD=true` directus crashes.



### To Reproduce

- Enable `EXTENSIONS_AUTO_RELOAD` in config
- Create a directory ouside of the project named `foo` with inside a file `index.js` with following content :
```javascript
import { defineHook } from '@directus/extensions-sdk'
export default defineHook(({ filter, action }) => {
  console.log('I\'m a hook!')
  filter('items.create', () => {
    console.log('Creating Item!')
  })

  action('items.create', () => {
    console.log('Item created!')
  })
})
```
- Start directus with `npx directus start`
- Move the `foo` directory to `extensions/hooks/`
- Notice in the console that the `foo` extension was loaded : `[22:20:29.737] INFO: Added extensions: foo`
- Delete the `foo` directory
- Save a file to trigger the file watcher
- Crash directus with following error message :
```javascript
[22:20:29.691] INFO: Reloading extensions
[22:20:29.737] INFO: Added extensions: foo
[22:20:56.851] INFO: Reloading extensions
node:internal/modules/cjs/loader:1075
  const err = new Error(message);
              ^

Error: Cannot find module '/workspaces/ddf-tools/packages/ddf-bundle-store/extensions/hooks/foo/index.js'
Require stack:
- /workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at ExtensionManager.unregisterApiExtensions (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js:483:42)
    at ExtensionManager.unload (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js:190:20)
    at file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js:94:28
    at JobQueue.run (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/utils/job-queue.js:18:19)
    at JobQueue.enqueue (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/utils/job-queue.js:11:18)
    at ExtensionManager.reload (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js:90:26)
    at FSWatcher.<anonymous> (file:///workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js:218:38)
    at FSWatcher.emit (node:events:513:28) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/workspaces/ddf-tools/node_modules/.pnpm/@directus+api@13.1.0_@types+node@20.6.2/node_modules/@directus/api/dist/extensions.js'
  ]
}

```

### Directus Version

v10.6.2

### Hosting Strategy

Self-Hosted (Custom)

--- a/.changeset/lucky-tables-leave.md
+++ b/.changeset/lucky-tables-leave.md
@@ -1,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+Fixed an issue where removing a local extension with autoload true resulted in a crash.

--- a/api/src/utils/delete-from-require-cache.ts
+++ b/api/src/utils/delete-from-require-cache.ts
@@ -1,7 +1,13 @@
 import { createRequire } from 'node:module';
+import logger from '../logger.js';
 
 const require = createRequire(import.meta.url);
 
 export function deleteFromRequireCache(modulePath: string): void {
-	delete require.cache[require.resolve(modulePath)];
+	try {
+		const moduleCachePath = require.resolve(modulePath);
+		delete require.cache[moduleCachePath];
+	} catch (error) {
+		logger.warn(`Module cache not found for ${modulePath}, skipped cache delete.`);
+	}
 }

--- a/api/src/utils/delete-from-require-cache.ts
+++ b/api/src/utils/delete-from-require-cache.ts
@@ -8,6 +8,6 @@ export function deleteFromRequireCache(modulePath: string): void {
 		const moduleCachePath = require.resolve(modulePath);
 		delete require.cache[moduleCachePath];
 	} catch (error) {
-		logger.warn(`Module cache not found for ${modulePath}, skipped cache delete.`);
+		logger.trace(`Module cache not found for ${modulePath}, skipped cache delete.`);
 	}
 }

